apps/frontend/src/app/3_organisms/BorrowLoan/components/ExtendLoanForm/ExtendLoanForm.tsx

```diff
@@ -0,0 +1,394 @@
+  ErrorBadge,
+  ErrorLevel,
+  HealthBar,
+  Paragraph,
+  RadioButton,
+  SimpleTable,
+  SimpleTableRow,
+} from '@sovryn/ui';
+import { Decimal } from '@sovryn/utils';
+
+import { AmountRenderer } from '../../../../2_molecules/AmountRenderer/AmountRenderer';
+import { AssetRenderer } from '../../../../2_molecules/AssetRenderer/AssetRenderer';
+import { DatePicker } from '../../../../2_molecules/DatePicker/DatePicker';
+import { LoanItem } from '../../../../5_pages/BorrowPage/components/OpenLoansTable/OpenLoansTable.types';
+import { COLLATERAL_RATIO_THRESHOLDS } from '../../../../../constants/general';
+import { MINIMUM_COLLATERAL_RATIO_BORROWING_MAINTENANCE } from '../../../../../constants/lending';
+import { getTokenDisplayName } from '../../../../../constants/tokens';
+import { useMaintenance } from '../../../../../hooks/useMaintenance';
+import { useGetRBTCPrice } from '../../../../../hooks/zero/useGetRBTCPrice';
+import { translations } from '../../../../../locales/i18n';
+import { dateFormat } from '../../../../../utils/helpers';
+import { decimalic } from '../../../../../utils/math';
+import { useGetCollateralAssetPrice } from '../AdjustLoanForm/hooks/useGetCollateralAssetPrice';
+import { normalizeToken, renderValue } from './ExtendLoanForm.utils';
+import { CurrentLoanData } from './components/CurrentLoanData/CurrentLoanData';
+import { useExtendLoan } from './hooks/useExtendLoan';
+
+const pageTranslations = translations.fixedInterestPage.extendLoanDialog;
+
+type ExtendLoanFormProps = {
+  loan: LoanItem;
+  onSuccess: () => void;
+};
+
+export const ExtendLoanForm: FC<ExtendLoanFormProps> = ({ loan }) => {
+  const { checkMaintenance, States } = useMaintenance();
+  const dappLocked = checkMaintenance(States.FULLD2);
+
+  const [nextRolloverDate, setNextRolloverDate] = useState(
+    dayjs(loan.rolloverDate * 1000)
+      .add(1, 'day')
+      .unix(),
+  );
+  const [isCalendarVisible, setIsCalendarVisible] = useState(false);
+  const [collateralAssetPrice, setCollateralAssetPrice] = useState('0');
+
+  const debtToken = useMemo(
+    () => normalizeToken(loan.debtAsset.toLowerCase()),
+    [loan.debtAsset],
+  );
+  const collateralToken = useMemo(
+    () => normalizeToken(loan.collateralAsset.toLowerCase()),
+    [loan.collateralAsset],
+  );
+  const { price: rbtcPrice } = useGetRBTCPrice();
+
+  const [useCollateral, setUseCollateral] = useState(true);
+
+  const { depositAmount, handleSubmit } = useExtendLoan(
+    loan,
+    debtToken,
+    nextRolloverDate,
+    useCollateral,
+  );
+
+  const { borrowPriceUsd, collateralPriceUsd } = useGetCollateralAssetPrice(
+    debtToken,
+    collateralToken,
+  );
+
+  const collateralChange = useMemo(() => {
+    if (useCollateral) {
+      const collateralUsdPrice =
+        collateralToken === SupportedTokens.rbtc
+          ? rbtcPrice
+          : collateralPriceUsd;
+      const loanTokenUsdPrice =
+        debtToken === SupportedTokens.rbtc ? rbtcPrice : borrowPriceUsd;
```

Comment:
Use `useMemo` for `collateralUsdPrice` and `loanTokenUsdPrice` as you use it in 2 calculations.